h2o-genmodel/src/main/java/hex/genmodel/ModelMojoReader.java

```diff
@@ -183,7 +183,10 @@ private void readAll(final boolean readModelDescriptor) throws IOException {
     checkMaxSupportedMojoVersion();
     readModelData();
     if (readModelDescriptor) {
-      _model._modelDescriptor = new ModelDescriptorBuilder(_model).build();
+      final String algoName = readkv("algo");
```

Comment:
This fixes a bug where algoName and full algo name are filled with one value only - the fully qualified class name of the model. We **do have these values in the MOJO**, and thus we should use those. Also, the reader builder relies on these values to construct the proper MOJO reader. As this PR is merged, those will be used to compare the algorithm against a list of algorithms allowed for MOJO import in `Generic` class.

```
    "algo": "gbm",
    "algo_full_name": "Gradient Boosting Machine",
```